bosh-director/lib/bosh/director/rendered_job_templates_cleaner.rb

```diff
@@ -1,20 +1,33 @@
 module Bosh::Director
   class RenderedJobTemplatesCleaner
-    def initialize(instance_model, blobstore)
+    def initialize(instance_model, blobstore, logger)
       @instance_model = instance_model
       @blobstore = blobstore
+      @logger = logger
     end
 
     def clean
       @instance_model.stale_rendered_templates_archives.each do |archive|
-        @blobstore.delete(archive.blobstore_id)
+        if @blobstore.exists? archive.blobstore_id
```

Comment:
@dk, the blobstore client like dav-blobstore-client will not throw NotFound exception, instead, it will generally throw out a BlobstoreError exception and saying that "Could not delete object", from this generic error, we can not tell if this error is because of object not found or object deletion errors